### PR TITLE
[NET-0006] Add transaction schema

### LINKED ISSUE #6: NET-0006: Create transaction model
**DoD**
- [ ] Transaction model created.
- [ ] Should be able to handle info to act as a purchase, or as a sale.
- [ ] Info contained should be able to handle multiple items, each with their quantities, subtotal amount (quantity * unit price), discount, the discounted amount, 

**Suggested Schema**
TransactionRow Schema
```
$itemId: string (eventually an id for the actual inventory model when implemented),
$unitPrice: number, (can be different from the price listed in the actual item, but the reason should be noted in the remarks)
$quantity: number,
$subTotalAmount: number,
discount: number,
$discountedAmount: number,
remarks: string, (not required, unless unitPrice is different from listed inventory item price)
```

Transaction Schema
```
$status: string (pending approval, pending delivery, delivered, incomplete payment, paid, cancelled)
$date: string or datetime model (different from when this file is created),
$buyer: objectId,
$seller: objectId,
$type: string (purchase or sale),
$secondaryType: string (SI, PO, OR, DR, etc.),
$number: number (ie. SI #, PO #),
$rows: [TransactionRow],
$totalAmount: number,
discount: number,
paymentAmount: number,
withholdingTax: number,
paymentMethod: string,
remarks: string,
```

**Additional context**
Related Story:
As a user, I want to track transactions, and be able to access them.

### PR DESCRIPTION
**Pull request by allendomingo**
## Description
Resolves #6 

**Changelog:**
- [x] Add transaction schema
- [x] Add transaction swagger definition

**Screenshot/s:**
image

### CODE REVIEW

File: models/transaction.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		required: true,
+	},
+	remarks: {
+		type: String,
+    default: '',
+	},
+});
+
+const transactionSchema = new Schema({
+  status: { // ie. pending approval, pending delivery, delivered, incomplete payment, paid, cancelled
```
Comment (RayAdrian) [inline models/transaction.js]: I think we can add enums to specify what strings are expected

File: models/transaction.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		type: mongoose.Schema.Types.ObjectId,
+    ref: 'Customer',
+		required: true,
+	},
+	seller: {
+		type: mongoose.Schema.Types.ObjectId,
+    ref: 'Supplier',
+		required: true,
+	},
+	type: { // purchase or sale
```
Comment (RayAdrian) [inline models/transaction.js]: Here as well

File: models/transaction.js (hunk 1 above)
Comment (allendomingo) [reply models/transaction.js]: Good suggestion

**Review by RayAdrian — APPROVED** [review]
Looks good! 💯

**Pull request merged**